NeoML/include/NeoML/Dnn/DnnSimpleAPI.h

```diff
@@ -0,0 +1,512 @@
+NEOML_API CLayerWrapper<CEuclideanLossLayer> EuclideanLoss( float lossWeight = 1.0f );
+NEOML_API CLayerWrapper<CHingeLossLayer> HingeLoss( float lossWeight = 1.0f );
+NEOML_API CLayerWrapper<CSquaredHingeLossLayer> SquaredHingeLoss(
+	float lossWeight = 1.0f );
+NEOML_API CLayerWrapper<CBinaryFocalLossLayer> BinaryFocalLoss( 
+	float focalForce, float lossWeight = 1.0f );
+NEOML_API CLayerWrapper<CCenterLossLayer> CenterLoss( int numberOfClasses,
+	float classCentersConvergenceRate, float lossWeight = 1.0f );
+NEOML_API CLayerWrapper<CMultiHingeLossLayer> MultiHingeLoss( float lossWeight = 1.0f );
+NEOML_API CLayerWrapper<CMultiSquaredHingeLossLayer> MultiSquaredHingeLoss(
+	float lossWeight = 1.0f );
+
+NEOML_API CLayerWrapper<CPositionalEmbeddingLayer> PositionalEmbedding(
+	CPositionalEmbeddingLayer::TPositionalEmbeddingType type );
+
+NEOML_API CLayerWrapper<CSoftmaxLayer> Softmax(
+	CSoftmaxLayer::TNormalizationArea normalizationArea );
+NEOML_API CLayerWrapper<CAddToObjectLayer> AddToObject();
+
+// Special values for parameters in Reshape
+enum TTransformRule {
+	TR_Remainder = -1, // Shape inference, only one layer should have this value.
+	TR_Same = -2 // Rest dimension the same.
+};
+
+NEOML_API CLayerWrapper<CTransformLayer> Transform( int batchLength,
+	int batchWidth, int listSize, int height, int width, int depth, int channel );
+
+NEOML_API CLayerWrapper<CTransposeLayer> Transpose( TBlobDim d1, TBlobDim d2 );
+
+NEOML_API CLayerWrapper<CConcatBatchWidthLayer> ConcatBatchWidth();
+NEOML_API CLayerWrapper<CConcatHeightLayer> ConcatHeight();
+NEOML_API CLayerWrapper<CConcatWidthLayer> ConcatWidth();
+NEOML_API CLayerWrapper<CConcatDepthLayer> ConcatDepth();
+NEOML_API CLayerWrapper<CConcatChannelsLayer> ConcatChannels();
+
+NEOML_API CLayerWrapper<CBatchNormalizationLayer> BatchNormalization(
+	bool isChannelBased, bool isZeroFreeTerm = false, float slowConvergenceRate = 1.0f );
+
+NEOML_API CLayerWrapper<CObjectNormalizationLayer> ObjectNormalization(
+	float epsilon = 1e-5f );
+
+NEOML_API CLayerWrapper<CMultiheadAttentionLayer> MultiheadAttention(
+	int headCount, int hiddenSize, int outputSize, float dropoutRate );
+
+NEOML_API CLayerWrapper<CMatrixMultiplicationLayer> MatrixMultiplication();
+
+NEOML_API CLayerWrapper<CEltwiseSumLayer> Sum();
+NEOML_API CLayerWrapper<CEltwiseMulLayer> Mul();
+NEOML_API CLayerWrapper<CEltwiseNegMulLayer> NegMul();
+NEOML_API CLayerWrapper<CEltwiseMaxLayer> Max();
+
+NEOML_API CLayerWrapper<CGlobalMaxPoolingLayer> GlobalMaxPooling( int maxCount );
+NEOML_API CLayerWrapper<CGlobalMeanPoolingLayer> GlobalMeanPooling();
+
+NEOML_API CLayerWrapper<CLstmLayer> Lstm(
+	int hiddenSize, float dropoutRate, bool isInCompatibilityMode = false );
+
+NEOML_API CLayerWrapper<CGruLayer> Gru( int hiddenSize );
+
+NEOML_API CLayerWrapper<CSubSequenceLayer> SubSequence(
+	int startPos, int length );
+// CSubSequenceLayer with SetReverse()
+NEOML_API CLayerWrapper<CSubSequenceLayer> Reverse();

```

Comment:
Why separate `Reverse()`. Isn't name too short and isn't obvious. Mb bool flag in `SubSequence` or `ReverseSequence`